shared/src/main/scala/crystal/AppRootContext.scala

```diff
@@ -0,0 +1,35 @@
+package crystal
+
+import cats.FlatMap
+import cats.implicits._
+import cats.effect.SyncIO
+import cats.effect.LiftIO
+import cats.effect.concurrent.Ref
+import cats.Functor
+
+trait AppRootContext[C] {
+  private val context: Ref[SyncIO, Option[C]] = Ref.unsafe[SyncIO, Option[C]](none[C])
```

Comment:
I suppose this is what you mean by the unsafe bits? I guess is no so different to use a `var` which maybe the default solution. Could we let implementors define it?

Reply:
I was actually referring to the `eval` method, which we agreed to change to `withCtx`.

I'm not 100% about letting implementors define the `Ref`, since the logic relies on it being initialized to `none[C]`. Implementors would have to respect this.

Reply:
ok I see, we can put this implementation detail here then, and in that case being `private` sounds best